internal/orchestrator/updates.go

```diff
@@ -132,7 +135,21 @@ func (o *Orchestrator) startConditionWatchers(ctx context.Context,
 					watcher.Stop()
 				case entry := <-watcher.Updates():
 					if entry == nil {
+						if sawNil {
```

Comment:
can this be covered by a test, by any chance?

Reply:
Can't cover this by a test, we'd have to engineer a broken connection to NATS, which is not straightforward. In general it is hard to test error handling at the NATS interface because engineering the correct fault is not possible.

I feel like moving this into a capture in this block actually detracts from readability.

Reply:
I tend to prefer smaller blocks, but this isn't a blocker so get this merged.